llvm/include/llvm/CodeGen/GlobalISel/GenericMachineInstrs.h

```diff
@@ -34,6 +34,22 @@ class GenericMachineInstr : public MachineInstr {
   static bool classof(const MachineInstr *MI) {
     return isPreISelGenericOpcode(MI->getOpcode());
   }
+
+  bool hasPoisonGeneratingFlags() const {
+    return getFlags() & (NoUWrap | NoSWrap | IsExact | Disjoint | NonNeg |
+                         FmNoNans | FmNoInfs);
+  }
+
+  void dropPoisonGeneratingFlags() {
+    clearFlag(NoUWrap);
+    clearFlag(NoSWrap);
+    clearFlag(IsExact);
+    clearFlag(Disjoint);
+    clearFlag(NonNeg);
+    clearFlag(FmNoNans);
+    clearFlag(FmNoInfs);
```

Comment:
Should be able to do this at once as well

Reply:
I don't see a good way to, `clearFlag` takes a `MIFlag` and checks its value, so this would require doing a bitwise-and with `getFlags()` within a `setFlags()` call which may still not work and will likely be much uglier.

Reply:
You can add a clearFlags to MachineInstr to bulk clear flags

Reply:
Done.